test/helpers/user.go

```diff
@@ -0,0 +1,190 @@
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apache/cloudstack-go/v2/cloudstack"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/pkg/cloud"
+)
+
+const TempUserName = "TemporaryUser"
+
+// GetDomainByPath fetches a domain by its path.
+func GetDomainByPath(csClient *cloudstack.CloudStackClient, path string) (string, error, bool) {
+	// Split path and get name.
+	path = strings.Trim(path, "/")
+	tokens := strings.Split(path, "/")
+
+	// Ensure the path begins with ROOT.
+	if !strings.EqualFold(tokens[0], "ROOT") {
+		tokens = append([]string{"ROOT"}, tokens...)
+	} else {
+		tokens[0] = "ROOT"
+	}
+	path = strings.Join(tokens, "/")
+
+	// Set present search/list parameters.
+	p := csClient.Domain.NewListDomainsParams()
+	p.SetListall(true)
+
+	// If path was provided also use level narrow the search for domain.
+	if level := len(tokens) - 1; level >= 0 {
+		p.SetLevel(level)
+	}
+
+	if resp, err := csClient.Domain.ListDomains(p); err != nil {
+		return "", err, false
+	} else {
+		for _, domain := range resp.Domains {
+			if domain.Path == path {
+				return domain.Id, nil, true
+			}
+		}
+	}
+
+	return "", nil, false
+}
+
+// CreateDomainUnderParent creates a domain as a sub-domain of the passed parent.
+func CreateDomainUnderParent(csClient *cloudstack.CloudStackClient, parentID string, domainName string) (string, error) {
+	p := csClient.Domain.NewCreateDomainParams(domainName)
+	p.SetParentdomainid(parentID)
+	resp, err := csClient.Domain.CreateDomain(p)
+	if err != nil {
+		return "", err
+	}
+	return resp.Id, nil
+}
+
+// GetOrCreateDomain gets or creates a domain as specified in the passed domain object.
+func GetOrCreateDomain(csClient *cloudstack.CloudStackClient, domain *cloud.Domain) error {
```

Comment:
is there any configuration on the cloudstack environment itself for this to work? Do domains/users have resource limits by default which need to be adjusted in anyway?

Reply:
The tests all pass for me and I did not do any special configuration.

Could you please try running these tests to see if the new integ account/domain stuff works on your end? I think it should. Would just be `gingko -v ./pkg/...` so long as your setup matches the config in the e2e directory.

Reply:
I got 48 tests passing with a bunch skipped like

```
S [SKIPPED] [0.001 seconds]
AffinityGroup Unit Tests
/Users/dribinm/workspaces/eksa/cluster-api-provider-cloudstack/pkg/cloud/affinity_groups_test.go:30
  AffinityGroup Integ Tests [BeforeEach]
  /Users/dribinm/workspaces/eksa/cluster-api-provider-cloudstack/pkg/cloud/affinity_groups_test.go:72
    Creates an affinity group.
    /Users/dribinm/workspaces/eksa/cluster-api-provider-cloudstack/pkg/cloud/affinity_groups_test.go:82

  Could not connect to ACS instance.
  In [BeforeEach] at: /Users/dribinm/workspaces/eksa/cluster-api-provider-cloudstack/pkg/cloud/affinity_groups_test.go:74
```
Any ideas how to resolve this? Some more logging would be helpful :)